core/src/com/unciv/logic/automation/BarbarianAutomation.kt

```diff
@@ -80,18 +81,36 @@ class BarbarianAutomation(val civInfo: CivilizationInfo) {
     private fun automateScout(unit: MapUnit) {
         val unitActions = UnitActions().getUnitActions(unit, UncivGame.Current.worldScreen)
         val unitDistanceToTiles = unit.movement.getDistanceToTiles()
-        val nearEnemies = battleHelper.getAttackableEnemies(unit, unitDistanceToTiles)
+        val nearEnemyTiles = battleHelper.getAttackableEnemies(unit, unitDistanceToTiles)
 
         // 1 - heal or run if death is near
         if (unit.health < 50) {
-            val possibleDamage = nearEnemies.map {
-                battleDamage.calculateDamageToAttacker(MapUnitCombatant(unit),
-                        Battle(unit.civInfo.gameInfo).getMapCombatantOfTile(it.tileToAttack)!!)
+            val possibleDamage = nearEnemyTiles
+                    .map {
+                        battleDamage.calculateDamageToAttacker(MapUnitCombatant(unit),
+                                Battle(unit.civInfo.gameInfo).getMapCombatantOfTile(it.tileToAttack)!!)
+                    }
+                    .sum()
+            val possibleHeal = unit.rankTileForHealing(unit.currentTile)
+            if (possibleDamage > possibleHeal || nearEnemyTiles.size > 0) {
+                // run
+                val possibleTiles = unitDistanceToTiles.filter { unit.movement.canMoveTo(it.key) }.keys
+                val enemies = nearEnemyTiles.mapNotNull { it.tileToAttack.militaryUnit }
+                var mostFartherTile: Pair<TileInfo, Float> = possibleTiles.random() to 0f
```

Comment:
FurthestTile
And this looks like a classic situation to use possibleTiles.maxBy {enemy.movement.getMovementCostBetweenAdjacentTiles(enemy.currentTile, tile, enemy.civInfo)}

Reply:
but what if we have many enemies
we must try to escape as far as possible from at least one enemy